mmdet/apis/train.py

```diff
@@ -68,6 +68,51 @@ def set_random_seed(seed, deterministic=False):
         torch.backends.cudnn.benchmark = False
 
 
+def scale_lr(cfg, logger):
+    """Automatically scaling LR according to GPU number and sample per GPU.
+
+    Args:
+        cfg (config): training config.
+        logger (logging.Logger): logger.
+    """
+    enable_auto_scale_lr = cfg.get('enable_auto_scale_lr', False)
+    if enable_auto_scale_lr is False:
+        logger.info('You disabled automatically scaling LR.')
+        return
+
+    # default batch size of pre-train model
+    original_batch_size = cfg.get('default_batch_size', None)
+    assert original_batch_size is not None
+
+    # calculate the batch size
+    gpu_number = len(cfg.gpu_ids)
+    batch_size = gpu_number * cfg.data.samples_per_gpu
+
+    logger.info(f'You are using {gpu_number} GPU(s) '
+                f'and {cfg.data.samples_per_gpu} samples per gpu '
+                f'means the batch is {batch_size}.')
+
+    if batch_size != original_batch_size:
+        # Get original LR from config
+        original_lr = cfg.optimizer.get('lr', 0)
+        assert original_lr != 0
+
+        # scale LR according to paper
+        # [linear scaling rule](https://arxiv.org/abs/1706.02677)
```

Comment:
```python
        # scale LR with linear scaling rule https://arxiv.org/abs/1706.02677
```
will be sufficient.

Reply:
Done

Reply:
Markdown is invalid (as far as I know).
This comment can be written in one line.

Reply:
mmdet basically follows pep8. https://github.com/open-mmlab/mmdetection/blob/master/setup.cfg
`Limit all lines to a maximum of 79 characters.` https://peps.python.org/pep-0008/#maximum-line-length